packages/graph/hash_graph/lib/graph/src/store/postgres/mod.rs

```diff
@@ -70,34 +80,115 @@ impl<K> Default for DependencyMap<K> {
 
 impl<K> DependencyMap<K>
 where
-    K: Eq + Hash + Clone,
+    K: Eq + Hash + Clone + Debug,
 {
     /// Inserts a dependency into the map.
     ///
     /// If the dependency does not already exist in the dependency map, it will be inserted with the
-    /// provided `resolved_depth` and a reference to this dependency will be returned in order to
-    /// continue resolving it. In the case, that the dependency already exists, the
-    /// `resolved_depth` will be compared with depth used when inserting it before:
-    /// - If the previous `resolved_depth` was `None`, the dependency was not resolved yet and the
-    ///   value is returned
-    /// - If the new depth is higher, the depth will be updated and a reference to the dependency
-    ///   will be returned in order to keep resolving it
-    /// - Otherwise, `None` will be returned as no further resolution is needed
-    pub fn insert(
+    /// provided `new_resolve_depth` and `new_interval`. If the dependency was already resolved it
+    /// is checked whether the new resolve depth and interval are more general than the existing
+    /// resolve depth and interval. If they are, the existing resolve depth and interval are updated
+    /// to cover the new resolve depth and interval.
+    ///
+    /// If the traversed entry has to be resolved further, [`DependencyStatus::Unresolved`] is
+    /// returned with the new resolve depth and interval that the traversal should continue with.
+    pub fn update(
         &mut self,
         identifier: &K,
-        resolved_depth: GraphResolveDepths,
+        new_resolve_depth: GraphResolveDepths,
+        new_interval: TimeInterval<ProjectedTime>,
     ) -> DependencyStatus {
         match self.resolved.raw_entry_mut().from_key(identifier) {
             RawEntryMut::Vacant(entry) => {
-                entry.insert(identifier.clone(), resolved_depth);
-                DependencyStatus::Unresolved
+                entry.insert(
+                    identifier.clone(),
+                    (new_resolve_depth, new_interval.clone()),
+                );
+                DependencyStatus::Unresolved(new_resolve_depth, new_interval)
             }
             RawEntryMut::Occupied(entry) => {
-                if entry.into_mut().update(resolved_depth) {
-                    DependencyStatus::Unresolved
-                } else {
+                let (current_depths, current_interval) = entry.into_mut();
+                let old_interval = current_interval.clone();
+                *current_interval = current_interval.clone().merge(new_interval.clone());
```

Comment:
Is the `merge` safe to use here (rather than `union`)? 

My guess is that we aren't confident right now that the intervals will be contiguous, so this feels like it would return subtly incorrect information.

Reply:
`merge` cannot fail. If the intervals do not overlap or are not adjacent, the points in between will also be included. 
pseudo code:
```
merged = (min(lhs.lower, rhs.lower), max(lhs.upper, rhs.upper))
```

Reply:
I know which is why I said about them being contiguous and returning incorrect information. And also is why I spoke about `union` :)

Reply:
This seems related to the discussion below about returning too much information being a bug or not